doc/user/methods/random_forest_regressor.md

```diff
@@ -0,0 +1,639 @@
+which can be used for custom behavior during learning. The full signature of 
+the class is as follow:
+
+```c++
+RandomForestRegressor<FitnessFunction,
+                      DimensionSelectionType,
+                      NumericSplitType,
+                      CategoricalSplitType,
+                      UseBootstrap>
+```
+
+ * `FitnessFunction` : the measure of goodness to use when deciding on tree
+   splits
+ * `DimensionSelectionType`: the strategy used for proposing dimensions to
+   attempt to split on
+ * `NumericSplitType`: the strategy used for finding splits on numeric data
+   dimensions
+ * `CategoricalSplitType`: the strategy used for finding splits on categorical
+   data dimensions
+ * `UseBootstrap`: a boolean indicating whether or not to use a bootstrap sample
+   when training each tree in the forest
+
+Below, details are given for the requirements of each of these template types.
+
+---
+
+#### `FitnessFunction`
+
+ * Specifies the fitness function to use when learning a decision tree.
+ * The `MSEGain` _(default)_ and `MADGain` classes are available for drop-in
+   usage.
+ * A custom class must implement three functions:
+
+```c++
+// You can use this as a starting point for implementation.
+class CustomFitnessFunction
+{
+  // Compute the gain for the given vector of values, where `values[i]` has an
+  // associated instance weight `weights[i]`.
+  //
+  // `RowType` and `WeightVecType` will be vector types following the Armadillo
+  // API.  If `UseWeights` is `false`, then the `weights` vector should be
+  // ignored (e.g. the responses are not weighted).
+  //
+  // In the version with `begin` and `end` parameters, only the subset between
+  // `labels[begin]` and `labels[end]` (inclusive) should be considered.
+  template<bool UseWeights, typename RowType, typename WeightVecType>
+  double Evaluate(const RowType& labels,
+                  const WeightVecType& weights);
+
+  template<bool UseWeights, typename RowType, typename WeightVecType>
+  double Evaluate(const RowType& labels,
+                  const WeightVecType& weights,
+                  const size_t begin,
+                  const size_t end);
+
+  // Return the output value for prediction for a leaf node whose training
+  // values are made up of the values in the vector `responses` (optionally with
+  // associated instance weights `weights`).
+  //
+  // `ResponsesType` and `WeightsType` will be vector types following the
+  // Armadillo API.  If `UseWeights` is `false`, then the `weights` vector
+  // should be ignored (e.g. the responses are not weighted).
+  template<bool UseWeights, typename ResponsesType, typename WeightsType>
+  double OutputLeafValue(const ResponsesType& responses,
+                         const WeightsType& weights);
+};
+```
+
+***Note:*** this API differs from the `FitnessFunction` API required for
+[`DecisionTree`](decision_tree.md)! <!-- TODO: fix link! -->
```

Comment:
I think this could be updated to the anchor: `decision_tree.md#fitnessfunction`.  Actually the same is true for `decision_tree_regressor.md` too if you'd like to update it.